app/api.py

```diff
@@ -82,13 +74,24 @@ def create_warrior():
     try:
         cursor.execute(sql, values)
         db.commit()
-        resp = Response(
-            response=json.dumps({"message": "Warrior created successfully"}),
-            status=201,
-        )
-        resp.headers["location"] = "/warrior/" + id
 
+        # Fetch the inserted record
+        cursor.execute("SELECT * FROM warriors WHERE id = %s", (id,))
+        inserted_warrior = cursor.fetchone()
+
+        # Construct the response
+        response_data = {
+            "id": inserted_warrior[0],
+            "name": inserted_warrior[1],
+            "dob": inserted_warrior[2],
+            "fight_skills": inserted_warrior[3].split(',')
+        }
```

Comment:
I don't think it's necessary to fetch the inserted record right after inserting it. The response data can be constructed with the request data.

Reply:
I think response_data should be blank because the POST request response body does not need to return data. Looking at your pytest assertions, you're looking for the warrior id. You can get that from the location header.

Reply:
removed the code that fetches the inserted record by post request. Modified the test function for extracting the id from the location header.